src/World_TEST.cc

```diff
@@ -691,3 +691,133 @@ TEST(DOMWorld, Plugins)
   world.ClearPlugins();
   EXPECT_TRUE(world.Plugins().empty());
 }
+
+/////////////////////////////////////////////////
+TEST(DOMWorld, LoadEarthSurface)
+{
+  sdf::World world;
+
+  std::ostringstream stream;
+  stream
+    << "<?xml version='1.0'?>"
+    << "<sdf version='1.10'>"
+    << "<world name='spherical coordinates'>"
+    << "<spherical_coordinates>"
+    << " <surface_model>EARTH_WGS84</surface_model>"
+    << "  <world_frame_orientation>ENU</world_frame_orientation>"
+    << "  <latitude_deg>-22.9</latitude_deg>"
+    << "  <longitude_deg>-43.2</longitude_deg>"
+    << "  <elevation>0</elevation>"
+    << "  <heading_deg>0</heading_deg>"
+    << "</spherical_coordinates>"
+    << "</world>"
+    << "</sdf>";
+
+  sdf::SDFPtr sdfParsed(new sdf::SDF());
+  sdf::init(sdfParsed);
+  ASSERT_TRUE(sdf::readString(stream.str(), sdfParsed));
+  auto SphericalCoordinatesSDF =
+    sdfParsed->Root()->GetElement("world");
+  ASSERT_NE(nullptr, SphericalCoordinatesSDF);
+
+  auto errors = world.Load(SphericalCoordinatesSDF);
```

Comment:
nit: I think it would be more compact to use `Root::LoadSdfString` and then get a world pointer from the loaded `sdf::Root`

https://github.com/gazebosim/sdformat/blob/6424ffa6e24cc5a29bbdaa9bd4dbb1908c44abf6/src/Root_TEST.cc#L69-L87

this comment applied to the following new tests as well

Reply:
Done https://github.com/gazebosim/sdformat/pull/1050/commits/21d0508055b5130a854b936e4efe39fd5136cc4a